src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -72,37 +72,68 @@ internal T WithHttpManager(IHttpManager httpManager)
         /// <summary>
         /// Sets the logging callback. For details see https://aka.ms/msal-net-logging
         /// </summary>
+        /// <param name="loggingCallback"></param>
+        /// <param name="logLevel">Desired level of logging.  The default is LogLevel.Info</param>
+        /// <param name="enablePiiLogging">Boolean used to enable/disable logging of
+        /// Personally Identifiable Information (PII).
+        /// PII logs are never written to default outputs like Console, Logcat or NSLog
+        /// Default is set to <c>false</c>, which ensures that your application is compliant with GDPR.
+        /// You can set it to <c>true</c> for advanced debugging requiring PII
+        /// </param>
+        /// <param name="enableDefaultPlatformLogging">Flag to enable/disable logging to platform defaults.
+        /// In Desktop/UWP, Event Tracing is used. In iOS, NSLog is used.
+        /// In android, logcat is used. The default value is <c>false</c>
+        /// </param>
         /// <returns>The builder to chain the .With methods</returns>
         /// <exception cref="InvalidOperationException"/> is thrown if the loggingCallback
         /// was already set on the application builder
-        public T WithLoggingCallback(LogCallback loggingCallback)
+        public T WithLogging(
+            LogCallback loggingCallback, 
+            LogLevel? logLevel = null, 
+            bool? enablePiiLogging = null, 
+            bool? enableDefaultPlatformLogging = null)
         {
             if (Config.LoggingCallback != null)
             {
                 throw new InvalidOperationException(CoreErrorMessages.LoggingCallbackAlreadySet);
             }
 
             Config.LoggingCallback = loggingCallback;
```

Comment:
We have tried as a policy to not throw in the individual "with" methods and having a null callback at validate is valid since it means we won't ahve one.  I would vote to not throw in this case.

---
In reply to: [252130116](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/851#discussion_r252130116) [](ancestors = 252130116)

Reply:
I'm ok not throwing exceptions (and I can see you from the general settings object have a place where you call this with null) not super happy about letting developers set something that doesn't make sense. what about at least adding something to the log informing them that what ever they did it will not have any effect. This will greatly help supportability. The log could include a link to the page where we explain how to setup logging.

---
In reply to: [252415577](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/851#discussion_r252415577) [](ancestors = 252415577,252130116)

Reply:
Also, does the developer need to provide a callback if `enableDefaultPlatformLogging` is `true`? 
maybe this setting needs to be separate?